pkg/sqlgen/sqlgen.go

```diff
@@ -206,8 +229,15 @@ func (t *MetaTable) UpsertDML(values []interface{}) (string, error) {
 			buf.WriteString(",")
 		}
 
-		val, err := maybeDecodeBase64(val,
-			isBase64EncodedFieldType(t.Fields[i].FieldType))
+		var err error
+		ft := t.Fields[i].FieldType
+		switch {
+		case isBase64EncodedFieldType(ft):
```

Comment:
To me it looks like this is doing two very different things here that could be a bit confusing. The base64 decode is to decode input, where-as the string handling is for output encoding. What are your thoughts on pushing the encoding down into SQLQuote where most of that logic is already encapsulated? It seems like we'd end up with less duplication and also reduce the chance that someone will end up using SQLQuote elsewhere and fall into the same trap. I also realize this could cause a crisis with the tests, so want to be sensitive to that as well. If that's the case, might make sense to leave a comment on SQLQuote for the exception case where nulls will be handled differently between MySQL and SQLite.

Reply:
+1 good idea. i think the tests will be fine since we're at least excluding sqlite3 from a test on the db executive that exercises this path (it's commented, too).